File: code/smv-debugger/src/test/java/org/fbme/smvDebugger/commons/ArrayUtils_Test.java
package org.fbme.smvDebugger.commons;

import org.junit.Assert;
import org.junit.Test;

public class ArrayUtils_Test {
  @Test
  public void test_toArray() {
    Assert.assertArrayEquals(new Object[]{1}, ArrayUtils.toArray(1));
  }

  @Test
  public void test_to2dArray() {
    final Object[][] result = ArrayUtils.to2dArray(new Object[]{1, 2});
    Assert.assertArrayEquals(new Object[]{1}, result[0]);
    Assert.assertArrayEquals(new Object[]{2}, result[1]);
  }

  @Test
  public void test_concatObjects() {
    Assert.assertArrayEquals(new Object[]{1, 2, 3}, ArrayUtils.concat(1, new Object[]{2, 3}));
  }

  @Test
  public void test_concatStrings() {
    Assert.assertArrayEquals(new String[]{"1", "2", "3"}, ArrayUtils.concat(new String[]{"1", "2"}, "3"));
  }

  @Test
  public void test_subarray() {
    Assert.assertArrayEquals(new String[]{"2", "3"}, ArrayUtils.subarray(new String[]{"1", "2", "3", "4"}, 1, 3));
  }
}

File: code/smv-debugger/src/test/kotlin/org/fbme/smvDebugger/commons/ArrayUtils_Test.kt
package org.fbme.smvDebugger.commons

import org.fbme.smvDebugger.commons.ArrayUtils.concat
import org.fbme.smvDebugger.commons.ArrayUtils.subarray
import org.fbme.smvDebugger.commons.ArrayUtils.to2dArray
import org.fbme.smvDebugger.commons.ArrayUtils.toArray
import org.junit.Assert
import org.junit.Test

class ArrayUtils_Test {
    @Test
    fun test_toArray() {
        Assert.assertArrayEquals(arrayOf<Any>(1), toArray(1))
    }

    @Test
    fun test_to2dArray() {
        val result = to2dArray(arrayOf(1, 2))
        Assert.assertArrayEquals(arrayOf<Any>(1), result[0])
        Assert.assertArrayEquals(arrayOf<Any>(2), result[1])
    }

    @Test
    fun test_concatObjects() {
        Assert.assertArrayEquals(arrayOf<Any>(1, 2, 3), concat(1, arrayOf(2, 3)))
    }

    @Test
    fun test_concatStrings() {
        Assert.assertArrayEquals(arrayOf("1", "2", "3"), concat(arrayOf("1", "2"), "3"))
    }

    @Test
    fun test_subarray() {
        Assert.assertArrayEquals(arrayOf("2", "3"), subarray(arrayOf("1", "2", "3", "4"), 1, 3))
    }
}
